Makefile

```
@@ -25,7 +25,13 @@ SRC                   := $(GOPATH)/src/github.com/argoproj/argo-workflows
 # docker image publishing options
 IMAGE_NAMESPACE       ?= quay.io/argoproj
 DEV_IMAGE             ?= $(shell [ `uname -s` = Darwin ] && echo true || echo false)
-TARGET_PLATFORM := $(shell [ `uname -m` = arm64 ] && echo linux/arm64 || echo linux/amd64)
+PLATFORM ?= $(shell uname -m)
+ifeq ($(PLATFORM), x86_64)
+PLATFORM=amd64
+else ifeq ($(PLATFORM), arm)
+PLATFORM=arm64
+endif
+TARGET_PLATFORM := $(shell echo linux/${PLATFORM})
```

```suggestion
TARGET_PLATFORM       ?= $(shell [ `uname -m` = x86_64 ] && echo linux/amd64 || echo linux/amd64)
```

can't we just flip the conditional? also `aarch64` and `arm64` are [equivalent](https://stackoverflow.com/a/47274698/3431180).

(plus change to `?=` and align spacing)

otherwise it might be more resilient to use `go env GOARCH`:
```suggestion
TARGET_PLATFORM       ?= linux/$(shell go env GOARCH)
```

and that would match https://github.com/argoproj/argo-workflows/pull/12754#pullrequestreview-1982586495 et al